MidiSynthExample/src/com/mobileer/midisynthexample/FakeKeyGenerator.java

```diff
@@ -40,18 +44,20 @@ public void run() {
         }
     };
 
+    public static FakeKeyGenerator getInstance() {
+        if (mInstance == null) {
+            mInstance = new FakeKeyGenerator();
+        }
+        return mInstance;
+    }
```

Comment:
It may be a good idea to wrap this method in a mutex just in case it's called from two different threads.

See [here](https://github.com/google/walt/blob/master/android/WALT/app/src/main/java/org/chromium/latency/walt/SimpleLogger.java#L40) for how I did it.